### PR TITLE
[SofaNG] Setup SceneUtility

### PR DESCRIPTION
**Pull request by fredroy**
#1527
based on 
- #2582

~~It was supposed to be called Utils, but I did not really like this name (plural+shorten name) and Utility was more appealing...~~
EDIT: called `SceneUtility` now

Deprecates partially *SofaBaseUtils* and *SofaGraphComponent*.

Questions:
### RequiredPlugin
 is needed to be implcitely loaded somewhere if a scene wants to load stuff.
Should it be moved in SofaCore (thus always available) or force load Sofa.Component.Utility somewhere
### SceneChecker
SceneChecker stuff is in SofaBaseUtils but they are not component tho so it does not make sense to move them into Sofa.Component. Should they move somewhere (in the framework like SimulationCore) ?

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests]

File: modules/SofaGraphComponent/SofaGraphComponent_test/SceneChecker_test.cpp [hunk 1]
```diff
@@ -82,13 +82,20 @@ struct SceneChecker_test : public BaseSimulationTest
         sofa::component::initSofaBaseUtils();
         sofa::component::initSofaBaseMechanics();
         sofa::component::initSofaMeshCollision();
+
+        sofa::helper::system::PluginManager::getInstance().getPluginMap();
+        auto& map = PluginManager::getInstance().getPluginMap();
+        for (const auto& elem : map)
+        {
+            std::cout <<  elem.second.getModuleName() << std::endl;
+        }
```
Comment (alxbilger) [inline modules/SofaGraphComponent/SofaGraphComponent_test/SceneChecker_test.cpp]: Is this important? You just `cout` something. Or is there side effects?

**Review by alxbilger — CHANGES_REQUESTED** [review]
Nice! I only have a little remark in a test.
I used https://github.com/fredroy/sofa/compare/sofang_io...sofang_utils to review your PR.

File: modules/SofaGraphComponent/SofaGraphComponent_test/SceneChecker_test.cpp (hunk 1 above)
Comment (fredroy) [reply modules/SofaGraphComponent/SofaGraphComponent_test/SceneChecker_test.cpp]: well spotted, it was for debug purpose and i did not remove this.. 👍

**Pull request merged**